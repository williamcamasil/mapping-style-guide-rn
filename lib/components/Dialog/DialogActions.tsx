import React from 'react';
import { View } from 'react-native';

import { useViewStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import Button, { ButtonPropsType } from '../Button';

type DialogTitlePropsType = {
  children?: React.ReactNode;
  theme: AppThemeType;
};

const throwErrorInDevMode = () => {
  if (__DEV__) {
    throw new Error('Dialog.Actions should be a Button element');
  }
};

const DialogActions: React.FC<DialogTitlePropsType> = ({ children, theme }) => {
  const footerStyles = useViewStyles(() => [
    {
      marginTop: theme.spacings.sXXL,
    },
  ], [theme.spacings.sXXL]);

  const mapChildren = (child: React.ReactNode, index: number) => {
    if (!React.isValidElement(child)) {
      return null;
    }

    if (child.type !== Button) {
      throwErrorInDevMode();

      return React.cloneElement(child, {
        key: index,
      });
    }

    const firstAction = index === 0;

    return React.cloneElement(child as React.ReactElement<ButtonPropsType>, {
      key: index,
      style: {
        marginTop: firstAction ? 0 : theme.spacings.sXS,
      },
      variant: firstAction ? 'containedPrimary' : 'text',
      size: firstAction ? 'large' : 'medium',
    });
  };

  const renderedActions = Array.isArray(children)
    ? children.map(mapChildren)
    : React.Children.map(children, mapChildren);

  if (!renderedActions?.length) return null;

  return (
    <View style={footerStyles}>
      {renderedActions}
    </View>
  );
};

export default withTheme(DialogActions);
