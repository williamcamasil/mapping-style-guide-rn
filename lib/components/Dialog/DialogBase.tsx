import React from 'react';
import {
  TouchableWithoutFeedback,
  View,
  StyleSheet,
  StyleProp,
  ViewStyle,
} from 'react-native';

import { useViewStyles } from '../../hooks';
import Modal, { ModalPropsType } from '../Modal';

export type DialogBasePropsType = ModalPropsType & {
  modalStyle?: StyleProp<ViewStyle>;
};

const styles = StyleSheet.create({
  modal: {
    flex: 1,
  },
  overlay: {
    flex: 1,
    backgroundColor: '#000000b3',
  },
  outsideTouch: {
    flex: 1,
  },
});

const DialogBase = ({
  children,
  modalStyle,
  dismissible,
  onDismiss,
  overlayStyle,
  ...others
}: DialogBasePropsType): React.ReactElement => {

  const modalStyles = useViewStyles(() => [
    styles.modal,
    modalStyle,
  ], [modalStyle]);

  const overlayStyles = useViewStyles(() => [
    styles.overlay,
    overlayStyle,
  ], [overlayStyle]);

  return (
    <Modal
      {...others}
      onDismiss={onDismiss}
      dismissible={dismissible}
      style={modalStyles}
      overlayStyle={overlayStyles}
      withSafeAreaIOS={false}
      skipStatusBarAndroid
    >
      <TouchableWithoutFeedback
        onPress={onDismiss}
        disabled={!dismissible}
        accessibilityLabel="Fechar janela"
        accessibilityRole="button"
      >
        <View style={styles.outsideTouch} />
      </TouchableWithoutFeedback>

      {children}
    </Modal>
  );
};

export default DialogBase;
