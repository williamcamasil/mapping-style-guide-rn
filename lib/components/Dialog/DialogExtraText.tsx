import React from 'react';
import { View } from 'react-native';

import { useTextStyles } from '../../hooks';
import { AppThemeType, withTheme } from '../../theme';
import Spacer from '../Spacer';
import Text from '../Text';

type DialogExtraTextPropsType = {
  children?: React.ReactNode;
  centered?: boolean;
  theme: AppThemeType;
};

const DialogExtraText: React.FC<DialogExtraTextPropsType> = ({ children, centered, theme }) => {
  const textStyles = useTextStyles(() => [
    {
      textAlign: centered ? 'center' : 'left',
    },
  ], [centered]);

  if (!children) return null;

  return (
    <View>
      <Spacer size={theme.spacings.sLarge} />
      <Text variant="body" color="neutralGray600" weight="bold" style={textStyles}>
        {children}
      </Text>
    </View>
  );
};

export default withTheme(DialogExtraText);
