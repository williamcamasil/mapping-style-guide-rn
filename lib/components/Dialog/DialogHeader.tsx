import React from 'react';
import { View } from 'react-native';

import { useViewStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';

type DialogHeaderPropsType = {
  children?: React.ReactNode;
  theme: AppThemeType;
};

const DialogHeader: React.FC<DialogHeaderPropsType> = ({ children, theme }) => {
  const headerStyles = useViewStyles(() => [
    {
      justifyContent: 'center',
      alignItems: 'center',
      marginBottom: theme.spacings.sXS,
    },
  ], [theme.spacings.sXS]);

  if (!children) return null;

  return (
    <View style={headerStyles}>
      {children}
    </View>
  );
};

export default withTheme(DialogHeader);
