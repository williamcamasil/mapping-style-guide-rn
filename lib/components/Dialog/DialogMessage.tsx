import React from 'react';

import { useTextStyles } from '../../hooks';
import Text from '../Text';

type DialogMessagePropsType = {
  children?: React.ReactNode;
  centered?: boolean;
};

const DialogMessage: React.FC<DialogMessagePropsType> = ({ children, centered }) => {
  const textStyles = useTextStyles(() => [
    {
      textAlign: centered ? 'center' : 'left',
    },
  ], [centered]);

  if (!children) return null;

  return (
    <Text variant="body" color="neutralGray600" style={textStyles}>
      {children}
    </Text>
  );
};

export default DialogMessage;
