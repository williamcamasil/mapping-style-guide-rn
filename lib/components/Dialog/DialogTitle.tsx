import React from 'react';

import { useTextStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import Text from '../Text';

type DialogTitlePropsType = {
  children?: React.ReactNode;
  centered?: boolean;
  theme: AppThemeType;
};

const DialogTitle: React.FC<DialogTitlePropsType> = ({ children, centered, theme }) => {
  const textStyles = useTextStyles(() => [
    {
      marginBottom: theme.spacings.sXXS,
      textAlign: centered ? 'center' : 'left',
    },
  ], [centered, theme.spacings.sXXS]);

  if (!children) return null;

  return (
    <Text variant="headingSmall" color="neutralGray700" style={textStyles}>
      {children}
    </Text>
  );
};

export default withTheme(DialogTitle);
