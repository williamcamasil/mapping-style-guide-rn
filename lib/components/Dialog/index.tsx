import React from 'react';
import {
  View,
  StyleSheet,
  StyleProp,
  ViewStyle,
  ScrollView,
  SafeAreaView,
} from 'react-native';

import { useViewStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import DialogActions from './DialogActions';
import DialogBase, { DialogBasePropsType } from './DialogBase';
import DialogExtraText from './DialogExtraText';
import DialogHeader from './DialogHeader';
import DialogMessage from './DialogMessage';
import DialogTitle from './DialogTitle';

export type DialogPropsType = DialogBasePropsType & {
  theme: AppThemeType;
  containerStyle?: StyleProp<ViewStyle>;
  dismissible?: boolean;
};

const styles = StyleSheet.create({
  container: {
    ...StyleSheet.absoluteFillObject,
    alignItems: 'center',
    justifyContent: 'center',
  },
  dialog: {
    maxWidth: 340,
    width: '100%',
  },
});

const Dialog = ({
  theme,
  children,
  style,
  containerStyle,
  ...others
}: DialogPropsType): React.ReactElement => {

  const dialogStyles = useViewStyles(() => [
    styles.dialog,
    {
      borderRadius: theme.borders.radius.XL,
      backgroundColor: theme.colors.neutralWhite,
      margin: theme.spacings.sNano,
    },
    style,
  ], [style, theme]);

  const scrollViewContentStyles = useViewStyles(() => [
    {
      padding: theme.spacings.sLarge,
    },
  ], [theme.spacings.sLarge]);

  const containerStyles = useViewStyles(() => [
    styles.container,
    containerStyle,
  ], [containerStyle]);

  return (
    <DialogBase
      {...others}
    >
      <SafeAreaView pointerEvents="box-none" style={containerStyles}>
        <View pointerEvents="box-none" style={dialogStyles}>
          <ScrollView
            contentContainerStyle={scrollViewContentStyles}
            bounces={false}
          >
            {children}
          </ScrollView>
        </View>
      </SafeAreaView>
    </DialogBase>
  );
};

Dialog.Title = DialogTitle;
Dialog.Header = DialogHeader;
Dialog.Message = DialogMessage;
Dialog.ExtraText = DialogExtraText;
Dialog.Actions = DialogActions;

Dialog.defaultProps = {
  dismissible: true,
};

export default withTheme(Dialog);
