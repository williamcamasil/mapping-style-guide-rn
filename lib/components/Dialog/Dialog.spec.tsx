import React from 'react';

import { render } from '@testing-library/react-native';

import Dialog from '.';
import Button from '../Button';
import Text from '../Text';
import SampleHeaderSvg from './sample-header.svg';

describe('Dialog snapshot', () => {
  it('default', () => {
    const tree = render(
      <Dialog
        isFocused
        visible
      >
        <Dialog.Header>
          <SampleHeaderSvg />
        </Dialog.Header>
        <Dialog.Title>
          Title
        </Dialog.Title>
        <Dialog.Message>
          Lorem ipsum is placeholder text commonly used in the graphic
        </Dialog.Message>
        <Dialog.ExtraText>
          Extra text
        </Dialog.ExtraText>
        <Dialog.Actions>
          <Button onPress={() => null}>
            Action 1
          </Button>
          <Button onPress={() => null}>
            Action 2
          </Button>
        </Dialog.Actions>
      </Dialog>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('not dismissible', () => {
    const tree = render(
      <Dialog
        isFocused
        visible
        dismissible={false}
      >
        <Dialog.Title>
          Title
        </Dialog.Title>
        <Dialog.Message>
          Lorem ipsum is placeholder text commonly used in the graphic
        </Dialog.Message>
      </Dialog>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('centered', () => {
    const tree = render(
      <Dialog
        isFocused
        visible
      >
        <Dialog.Title centered>
          Title
        </Dialog.Title>
        <Dialog.Message centered>
          Lorem ipsum is placeholder text commonly used in the graphic
        </Dialog.Message>
      </Dialog>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('invisible', () => {
    const tree = render(
      <Dialog
        isFocused={false}
        visible={false}
      >
        <Dialog.Title>
          Title
        </Dialog.Title>
        <Dialog.Message>
          Lorem ipsum is placeholder text commonly used in the graphic
        </Dialog.Message>
      </Dialog>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('empty', () => {
    const tree = render(
      <Dialog
        isFocused
        visible
      >
        <Dialog.Header />
        <Dialog.Message />
        <Dialog.Title />
        <Dialog.ExtraText />
        <Dialog.Actions />
      </Dialog>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('invalid actions', () => {
    (global as any).__DEV__ = true;

    expect(() => {
      render(
        <Dialog
          isFocused
          visible
        >
          <Dialog.Actions>
            <Text>
              Teste
            </Text>
          </Dialog.Actions>
        </Dialog>,
      );
    }).toThrowError();

    (global as any).__DEV__ = false;

    const result = render(
      <Dialog
        isFocused
        visible
      >
        <Dialog.Actions>
          <Text>
            Teste
          </Text>
          Teste
        </Dialog.Actions>
      </Dialog>,
    );

    expect(result).toBeDefined();
  });
});
