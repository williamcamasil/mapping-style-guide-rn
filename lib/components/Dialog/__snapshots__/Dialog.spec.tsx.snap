// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog snapshot centered 1`] = `
<View
  pointerEvents="box-none"
  style={
    {
      "backgroundColor": "transparent",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
    testID="modal-safeArea"
  >
    <View
      accessibilityLiveRegion="polite"
      accessibilityViewIsModal={true}
      collapsable={false}
      style={
        {
          "backgroundColor": "#000000b3",
          "flex": 1,
          "opacity": 0,
        }
      }
      testID="modal-overlay"
    >
      <View
        collapsable={false}
        style={
          {
            "flex": 1,
            "marginTop": undefined,
            "transform": [
              {
                "translateY": 1334,
              },
            ],
          }
        }
        testID="modal-content"
      >
        <View
          accessibilityLabel="Fechar janela"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "flex": 1,
            }
          }
        />
        <RCTSafeAreaView
          pointerEvents="box-none"
          style={
            {
              "alignItems": "center",
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 24,
                "margin": 8,
                "maxWidth": 340,
                "width": "100%",
              }
            }
          >
            <RCTScrollView
              bounces={false}
              contentContainerStyle={
                {
                  "padding": 28,
                }
              }
            >
              <View>
                <Text
                  allowFontScaling={false}
                  style={
                    {
                      "color": "#3E4149",
                      "flexShrink": 1,
                      "fontFamily": "Averta CY",
                      "fontSize": 23,
                      "fontWeight": "bold",
                      "letterSpacing": -0.02,
                      "lineHeight": 28,
                      "marginBottom": 12,
                      "textAlign": "center",
                    }
                  }
                >
                  Title
                </Text>
                <Text
                  allowFontScaling={false}
                  style={
                    {
                      "color": "#676D79",
                      "flexShrink": 1,
                      "fontFamily": "Averta CY",
                      "fontSize": 16,
                      "fontWeight": "normal",
                      "letterSpacing": -0.02,
                      "lineHeight": 22,
                      "textAlign": "center",
                    }
                  }
                >
                  Lorem ipsum is placeholder text commonly used in the graphic
                </Text>
              </View>
            </RCTScrollView>
          </View>
        </RCTSafeAreaView>
      </View>
    </View>
  </View>
</View>
`;

exports[`Dialog snapshot default 1`] = `
<View
  pointerEvents="box-none"
  style={
    {
      "backgroundColor": "transparent",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
    testID="modal-safeArea"
  >
    <View
      accessibilityLiveRegion="polite"
      accessibilityViewIsModal={true}
      collapsable={false}
      style={
        {
          "backgroundColor": "#000000b3",
          "flex": 1,
          "opacity": 0,
        }
      }
      testID="modal-overlay"
    >
      <View
        collapsable={false}
        style={
          {
            "flex": 1,
            "marginTop": undefined,
            "transform": [
              {
                "translateY": 1334,
              },
            ],
          }
        }
        testID="modal-content"
      >
        <View
          accessibilityLabel="Fechar janela"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "flex": 1,
            }
          }
        />
        <RCTSafeAreaView
          pointerEvents="box-none"
          style={
            {
              "alignItems": "center",
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 24,
                "margin": 8,
                "maxWidth": 340,
                "width": "100%",
              }
            }
          >
            <RCTScrollView
              bounces={false}
              contentContainerStyle={
                {
                  "padding": 28,
                }
              }
            >
              <View>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "marginBottom": 16,
                    }
                  }
                >
                  <SvgMock />
                </View>
                <Text
                  allowFontScaling={false}
                  style={
                    {
                      "color": "#3E4149",
                      "flexShrink": 1,
                      "fontFamily": "Averta CY",
                      "fontSize": 23,
                      "fontWeight": "bold",
                      "letterSpacing": -0.02,
                      "lineHeight": 28,
                      "marginBottom": 12,
                      "textAlign": "left",
                    }
                  }
                >
                  Title
                </Text>
                <Text
                  allowFontScaling={false}
                  style={
                    {
                      "color": "#676D79",
                      "flexShrink": 1,
                      "fontFamily": "Averta CY",
                      "fontSize": 16,
                      "fontWeight": "normal",
                      "letterSpacing": -0.02,
                      "lineHeight": 22,
                      "textAlign": "left",
                    }
                  }
                >
                  Lorem ipsum is placeholder text commonly used in the graphic
                </Text>
                <View>
                  <Text
                    style={
                      {
                        "height": 28,
                        "width": 28,
                      }
                    }
                  />
                  <Text
                    allowFontScaling={false}
                    style={
                      {
                        "color": "#676D79",
                        "flexShrink": 1,
                        "fontFamily": "Averta CY",
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "letterSpacing": -0.02,
                        "lineHeight": 22,
                        "textAlign": "left",
                      }
                    }
                  >
                    Extra text
                  </Text>
                </View>
                <View
                  style={
                    {
                      "marginTop": 40,
                    }
                  }
                >
                  <View
                    style={{}}
                    testID="container-button"
                  >
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "#013188",
                          "borderRadius": 12,
                          "borderWidth": 0,
                          "display": "flex",
                          "height": 64,
                          "justifyContent": "center",
                          "marginTop": 0,
                          "minWidth": 64,
                          "opacity": 1,
                          "paddingHorizontal": 32,
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        style={
                          {
                            "color": "#FFFFFF",
                            "flexShrink": 1,
                            "fontFamily": "Averta CY",
                            "fontSize": 16,
                            "fontWeight": "bold",
                            "letterSpacing": -0.02,
                            "lineHeight": 19,
                          }
                        }
                        testID="text-with-icon"
                      >
                        Action 1
                      </Text>
                    </View>
                  </View>
                  <View
                    style={{}}
                    testID="container-button"
                  >
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "borderRadius": 12,
                          "borderWidth": 0,
                          "display": "flex",
                          "height": 48,
                          "justifyContent": "center",
                          "marginTop": 16,
                          "minWidth": 48,
                          "opacity": 1,
                          "paddingHorizontal": 24,
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        style={
                          {
                            "color": "#A4A7AF",
                            "flexShrink": 1,
                            "fontFamily": "Averta CY",
                            "fontSize": 16,
                            "fontWeight": "bold",
                            "letterSpacing": -0.02,
                            "lineHeight": 19,
                          }
                        }
                        testID="text-with-icon"
                      >
                        Action 2
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </RCTSafeAreaView>
      </View>
    </View>
  </View>
</View>
`;

exports[`Dialog snapshot empty 1`] = `
<View
  pointerEvents="box-none"
  style={
    {
      "backgroundColor": "transparent",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
    testID="modal-safeArea"
  >
    <View
      accessibilityLiveRegion="polite"
      accessibilityViewIsModal={true}
      collapsable={false}
      style={
        {
          "backgroundColor": "#000000b3",
          "flex": 1,
          "opacity": 0,
        }
      }
      testID="modal-overlay"
    >
      <View
        collapsable={false}
        style={
          {
            "flex": 1,
            "marginTop": undefined,
            "transform": [
              {
                "translateY": 1334,
              },
            ],
          }
        }
        testID="modal-content"
      >
        <View
          accessibilityLabel="Fechar janela"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "flex": 1,
            }
          }
        />
        <RCTSafeAreaView
          pointerEvents="box-none"
          style={
            {
              "alignItems": "center",
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 24,
                "margin": 8,
                "maxWidth": 340,
                "width": "100%",
              }
            }
          >
            <RCTScrollView
              bounces={false}
              contentContainerStyle={
                {
                  "padding": 28,
                }
              }
            >
              <View />
            </RCTScrollView>
          </View>
        </RCTSafeAreaView>
      </View>
    </View>
  </View>
</View>
`;

exports[`Dialog snapshot invisible 1`] = `null`;

exports[`Dialog snapshot not dismissible 1`] = `
<View
  pointerEvents="box-none"
  style={
    {
      "backgroundColor": "transparent",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
    testID="modal-safeArea"
  >
    <View
      accessibilityLiveRegion="polite"
      accessibilityViewIsModal={true}
      collapsable={false}
      style={
        {
          "backgroundColor": "#000000b3",
          "flex": 1,
          "opacity": 0,
        }
      }
      testID="modal-overlay"
    >
      <View
        collapsable={false}
        style={
          {
            "flex": 1,
            "marginTop": undefined,
            "transform": [
              {
                "translateY": 1334,
              },
            ],
          }
        }
        testID="modal-content"
      >
        <View
          accessibilityLabel="Fechar janela"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "flex": 1,
            }
          }
        />
        <RCTSafeAreaView
          pointerEvents="box-none"
          style={
            {
              "alignItems": "center",
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 24,
                "margin": 8,
                "maxWidth": 340,
                "width": "100%",
              }
            }
          >
            <RCTScrollView
              bounces={false}
              contentContainerStyle={
                {
                  "padding": 28,
                }
              }
            >
              <View>
                <Text
                  allowFontScaling={false}
                  style={
                    {
                      "color": "#3E4149",
                      "flexShrink": 1,
                      "fontFamily": "Averta CY",
                      "fontSize": 23,
                      "fontWeight": "bold",
                      "letterSpacing": -0.02,
                      "lineHeight": 28,
                      "marginBottom": 12,
                      "textAlign": "left",
                    }
                  }
                >
                  Title
                </Text>
                <Text
                  allowFontScaling={false}
                  style={
                    {
                      "color": "#676D79",
                      "flexShrink": 1,
                      "fontFamily": "Averta CY",
                      "fontSize": 16,
                      "fontWeight": "normal",
                      "letterSpacing": -0.02,
                      "lineHeight": 22,
                      "textAlign": "left",
                    }
                  }
                >
                  Lorem ipsum is placeholder text commonly used in the graphic
                </Text>
              </View>
            </RCTScrollView>
          </View>
        </RCTSafeAreaView>
      </View>
    </View>
  </View>
</View>
`;
